sssd_test_framework/utils/pam.py

```diff
@@ -0,0 +1,235 @@
+
+    def access(self, file: str | None = None) -> PAMAccess:
+        """
+        :param file: PAM Access file name.
+        :type file: str, optional
+        :return: PAM Access object
+        :rtype: PAMAccess
+        """
+        if file is None:
+            return PAMAccess(self)
+
+        return PAMAccess(self, file)
+
+    def faillock(self, user: str | None = None, file: str | None = None) -> PAMFaillock:
+        """
+        :param user: Username.
+        :type user: str, optional
+        :param file: PAM Faillock file name.
+        :type file: str, optional
+        :return: PAM Faillock object
+        :rtype: PAMFaillock
+        """
+        if file is None:
+            return PAMFaillock(self, user)
+        if user is None:
+            return PAMFaillock(self, file)
+        if user and file is None:
+            return PAMFaillock(self)
+
+        return PAMFaillock(self, user, file)
+
+
+class PAMAccess:
+    """
+    Management of PAM Access on the client host.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+        def test_example(client: Client, provider: GenericProvider):
+            # Add users
+            provider.user("user-1").add()
+            provider.user("user-1").add()
+
+            # Add rule to permit "user-1" and deny "user-2"
+            client.pam.access.add(["+:user-1:ALL","-:user-2:NONE"])
+
+            client.sssd.common.pam(["with-pamaccess"])
+            client.sssd.start()
+
+            # Check the results
+            assert client.auth.ssh.password("user-1", "Secret123")
+            assert not client.auth.ssh.password("user-2", "Secret123")
+    """
+
+    def __init__(self, util: PAMUtils, file: str = "/etc/security/access.conf") -> None:
+        """
+        :param util: PAMUtils utility object
+        :type util: PAMUtils
+        :param file: File name of access file
+        :type file: str, optional
+        """
+        self.util: PAMUtils = util
+        self.file: str = file
+
+    def get(self) -> list[str]:
+        """
+        Get PAM rules from access file.
+        :return: List of PAM access rules
+        :rtype: list
+        """
+        result = self.util.fs.read(self.file).split("\n")
+        self.util.logger.info(f"{result} are in {self.file} on {self.util.host.hostname}")
+
+        return result
+
+    def add(self, rules: list[str]) -> PAMAccess:
+        """
+        Add one or more rules to access file.
+        :param rules: Access rule
+        :type rules: list[str], required
+        :return: self
+        :rtype: PAMAccess
+        """
+        content = ""
+        for i in rules:
+            content += i + "\n"
+
+        self.util.logger.info(f"{content} written to {self.file} on {self.util.host.hostname}")
+        self.util.fs.write(self.file, content, dedent=False)
+
+        return self
+
+    def delete(self, rules: list[str]) -> PAMAccess:
```

Comment:
See comment to add - if we change it to write, you can use this method instead of building `content` and calling fs.write